### PR TITLE
lib: use upstream functions and parse digest

### PR DESCRIPTION
**Pull request by iaguis**
Now that we vendor, we can get rid of some copied code.

Also, this might help with https://github.com/coreos/rkt/issues/2188

cc @dgonyeo

### CODE REVIEW

File: lib/internal/docker/docker_functions.go [hunk 1]
```diff
@@ -35,118 +35,21 @@ const (
...
-		remoteName = reposName
+// SplitReposName breaks a repo name into an index name and remote name.
+func SplitReposName(name string) (indexName, remoteName string) {
+	i := strings.IndexRune(name, '/')
+	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != "localhost") {
+		indexName, remoteName = defaultIndexURL, name
 	} else {
-		indexName = nameParts[0]
-		remoteName = nameParts[1]
+		indexName, remoteName = name[:i], name[i+1:]
```
Comment (krnowak) [inline lib/internal/docker/docker_functions.go]: This case will be visited also when `i` is `-1`, so you cannot use it for indexing, right?

Comment (krnowak) [inline lib/internal/docker/docker_functions.go]: This case will be visited also when `i` is `-1`, so you cannot use it for indexing, right?

Comment (iaguis) [inline lib/internal/docker/docker_functions.go]: if `i` is `-1` it will fall under the `if` condition, oder?

Comment (krnowak) [inline lib/internal/docker/docker_functions.go]: Eh, nevermind.

File: lib/internal/docker/docker_functions.go [hunk 2]
```diff
@@ -232,31 +135,27 @@ func GetAuthInfo(indexServer string) (string, string, error) {
...
-		return nil, fmt.Errorf("invalid Docker image reference %q", arg)
+	r, err := reference.ParseNamed(arg)
+	if err != nil {
+		return nil, err
 	}
 
-	taglessRemote, tag := parseRepositoryTag(arg)
-	if tag == "" {
-		tag = defaultTag
+	tag := "latest"
```
Comment (krnowak) [inline lib/internal/docker/docker_functions.go]: `tag := defaultTag`?

Comment (iaguis) [inline lib/internal/docker/docker_functions.go]: yep

**Comment by krnowak** [comment]
One nit. Otherwise LFAD.

**Pull request merged**